src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Windows.cs

```diff
@@ -207,15 +269,27 @@ internal unsafe SocketError DoOperationAccept(Socket socket, SafeCloseSocket han
         {
             PrepareIOCPOperation();
 
+            byte* bufferPtr;
+            int receiveDataLength = 0;
```

Comment:
I don't know that it matters at all, but you could move the initialization of receiveDataLength to 0 into the _buffer == null branch.